engine/table/src/main/java/io/deephaven/engine/table/impl/PushdownResult.java

```diff
@@ -45,30 +68,209 @@ public class PushdownResult implements SafeCloseable {
      */
     private final WritableRowSet maybeMatch;
 
-    private PushdownResult(
-            final WritableRowSet match,
-            final WritableRowSet maybeMatch) {
-        this.match = match;
-        this.maybeMatch = maybeMatch;
+    /**
+     * Constructs a new result with all of {@code selection} as {@link #maybeMatch() maybeMatch}.
+     *
+     * @param selection the selection
+     * @return the result
+     */
+    public static PushdownResult maybeMatch(final RowSet selection) {
```

Comment:
Can we change `selection` to `input` in these creation helpers? Confusing when we have a `selection` member.